### ISSUE TITLE
Add Annotation Regex to UnusedPrivateMember

### ISSUE THREAD

**Issue by inktomi** [detekt/detekt#5296]
## Expected Behavior of the rule

In Compose, it's useful to write methods like this:

```kotlin
@Preview
@Composable
fun MyFeatureDemoWithCertainState() {
 //
}
```

That is better made a private function - but doing so causes a violation of `UnusedPrivateMember`. The rule's configuration allows for a regex that is ran against the method name. This means that we could start all our preview functions with `Preview` or similar, and add a regex there - but since they're already annotated `@Preview` it would be better to allow unused private methods with that annotation instead. 

If we could add support for allowing methods marked with `@Preview` to not be flagged by `UnusedPrivateMember` it would solve this issue.

**Comment by 3flex** [comment]
I definitely prefer the annotation method - allowing specified annotations to skip this check on private methods.

**Comment by BraisGabin** [comment]
That already exists using [suppressors](https://detekt.dev/docs/introduction/suppressors):

```yml
    ignoreAnnotated:
      - 'Preview'
```

We also have a guide to adapt the default configuration to Compose: https://detekt.dev/docs/introduction/compose
